### ISSUE TITLE
Improve TestTaxonomyFacetAssociations#validateFloats to not rely on summation ordering

### ISSUE THREAD

**Issue by gsmiller** [apache/lucene#13738]
### Description

After merging #13726 we saw test failures because `TestTaxonomyFacetAssociations#validateFloats` is written to (intentionally) sum floats in a consistent order and then use exact equality, but the test update brought in search concurrency which breaks that consistency. This got fixed with 0ec453d485df5411d21edd24b47f4880befd30d0, which just disables the concurrency in these tests for now, but maybe we should make this test less fragile and bring the concurrency back? Could be a nice opportunity to leverage #13723 when it gets merged.

**Comment by javanna** [comment]
Sounds good to me, I disabled concurrency in that test because I could not come up with a quick way to keep it enabled for this test. It looked to me as if it requires sequential execution, but I am not an expert of facets.

**Comment by mikemccand** [comment]
We could also accept that operations are sometimes out of order and use the [new ULP-based epsilon float equality assertion method](https://github.com/apache/lucene/pull/13723) to check.

**Comment by gsmiller** [comment]
@javanna yeah, +1 to the quick fix you put in place (after all, the test itself documents that it expects sequential, consistent ordering in summation). @mikemccand, agreed.

I don't see any urgency to picking this up but wanted to capture the idea. Could be a good starter task for someone potentially.
